p2p/switch.go

```diff
@@ -547,13 +542,14 @@ func (sw *Switch) addPeer(pc peerConn) error {
 	peerID := peerNodeInfo.ID
 
 	// ensure connection key matches self reported key
-	if pc.config.AuthEnc {
-		connID := pc.ID()
-
-		if peerID != connID {
-			return fmt.Errorf("nodeInfo.ID() (%v) doesn't match conn.ID() (%v)",
-				peerID, connID)
-		}
+	connID := pc.ID()
+
+	if peerID != connID {
```

Comment:
We can do this before the handshake now !

Reply:
Not really as we need the `peerNodeInfo`, or is this available without doing the handshake?

Reply:
oh whoops. I meant some of the checks below. Like checking for connections to self, duplicate peer, filterconnbyid, etc.